tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py

```diff
@@ -133,8 +134,44 @@ def _on_run_sql_action(self, event: ActionEvent):
 
         event.set_results({"results": json.dumps(results)})
 
+    def _on_test_tls_action(self, event: ActionEvent):
+        """An action that allows us to run SQL queries from this charm."""
+        logger.info(event.params)
+
+        relation_id = event.params["relation-id"]
+        relation_name = event.params["relation-name"]
+        if relation_name == self.first_database.relation_name:
+            relation = self.first_database
+        elif relation_name == self.second_database.relation_name:
+            relation = self.second_database
+        else:
+            event.fail(message="invalid relation name")
+
+        databag = relation.fetch_relation_data()[relation_id]
+
+        dbname = event.params["dbname"]
+        user = databag.get("username")
+        password = databag.get("password")
+
+        if event.params["readonly"]:
+            host = databag.get("read-only-endpoints").split(",")[0]
+            dbname = f"{dbname}_readonly"
+        else:
+            host = databag.get("endpoints").split(",")[0]
+        endpoint = host.split(":")[0]
+        port = host.split(":")[1]
+
+        try:
+            connection = self.connect_to_database(
+                database=dbname, user=user, password=password, host=endpoint, port=port, tls=True
+            )
+            connection.close()
+            event.set_results({"results": "True"})
+        except psycopg2.OperationalError:
+            event.set_results({"results": "False"})
```

Comment:
Check if we can start a connection with required encryption and report accordingly.